### PR TITLE
Add APIConfig Struct

### PR DESCRIPTION
**Pull request by nwchandler**
This merge adds an `APIConfig` struct to collect the parameters required by an `APIClient`. The `NewAPIClient` function has been updated to use this new struct, and any callers of `NewAPIClient` have been updated to use an `APIConfig` structure.

`NewAPIClient` tests were updated to use this structure, but validation was not updated, so that we could verify that behavior remained consistent.

As a bit of cleanup, TLS config and setup were moved in the api client file, so that API concerns and TLS/HTTP Transport concerns were closer to similar code. Previously, they were interspersed.

### CODE REVIEW

**Comment by nwchandler** [comment]
Comment to kick off functional tests. 😄

File: client/apiclient.go [hunk 1]
```diff
@@ -18,41 +18,10 @@ type HTTPClient interface {
 	Do(*http.Request) (*http.Response, error)
 }
 
-// TLSConfig contains the parameters needed to configure TLS on the HTTP client
```
Comment (nwchandler) [inline client/apiclient.go:21]: FYI, the TLS stuff moved toward the bottom of the file. Sorry it makes this file kind of ugly to review.

File: client/apiclient.go [hunk 2]
```diff
@@ -18,41 +18,10 @@ type HTTPClient interface {
...
-	TLSServerName string
-
-	// Insecure enables or disables SSL verification. Setting to `true` is highly
-	// discouraged.
-	Insecure bool
-}
-
 // NewAPIClient gets an API client for a product.
-func NewAPIClient(product, baseURL string, headers map[string]string, tlsConfig TLSConfig) (*APIClient, error) {
+func NewAPIClient(cfg APIConfig) (*APIClient, error) {
```
Comment (nwchandler) [inline client/apiclient.go:22]: This is where the substantial changes in the file are.

File: client/apiclient.go [hunk 3]
```diff
@@ -65,99 +34,26 @@ func NewAPIClient(product, baseURL string, headers map[string]string, tlsConfig
...
-		}
-		tlsClientConfig, err := tc.tlsConfigFunction(tc.tlsConfig)
-		if err != nil {
-			return nil, err
-		}
-		transport.TLSClientConfig = tlsClientConfig
-	}
-	return transport, nil
-}
+// APIConfig contains the configuration details for an APIClient.
```
Comment (nwchandler) [inline client/apiclient.go:42]: This is also a primary change for the PR.

File: client/apiclient.go [hunk 4]
```diff
@@ -229,3 +125,113 @@ func (c *APIClient) request(method string, path string, data []byte) (interface{
 
 	return iface, err
 }
+
```
Comment (nwchandler) [inline client/apiclient.go:128]: Everything below here is just code that got moved from above, but there were no "new" changes.

**Review by mkcp — APPROVED** [review]
Just what we scoped out - appreciate the review notes on the core changes vs. some incidental moved functions, made reading through better!

**Pull request merged**